homeassistant/components/volvooncall/__init__.py

```diff
@@ -197,19 +197,21 @@ def discover_vehicle(self, vehicle):
     async def update(self):
         """Update status from the online service."""
         try:
-            if not await self.connection.update(journal=True):
-                return False
-
-            for vehicle in self.connection.vehicles:
-                if vehicle.vin not in self.vehicles:
-                    self.discover_vehicle(vehicle)
-
-            return True
+            update_successful = await self.connection.update(journal=True)
```

Comment:
Can `update_successful` be falsy without raising an exception?

Reply:
Indeed it can not...🤦‍♂️ Thank you, and fixing now.